PYIC-8327: Remove kidJarHeaderEnabled feature flag.

undermad opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Features Set - KID_JAR_HEADER removed and cleaned places that depends on it.

### Why did it change

As kidJarHeaderEnabled feature flag is enabled across all environments, it can be removed from the project.

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-8327](https://govukverify.atlassian.net/browse/PYIC-8327)
- [ipv-core-common-infra](https://github.com/govuk-one-login/ipv-core-common-infra/pull/1228)

## Checklists

<!-- Delete if changes don't apply -->
- [x] API/unit/contract tests have been written/updated


[PYIC-8327]: https://govukverify.atlassian.net/browse/PYIC-8327?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

undermad committed:
PYIC-8327: Remove kidJarHeaderEnabled feature flag.

As kidJarHeaderEnabled feature flag is enabled across all environments, it can be removed from the project.

AmritSidhu commented on libs/cri-api-service/src/main/java/uk/gov/di/ipv/core/library/criapiservice/CriApiService.java:
```diff
@@ -196,10 +195,7 @@ private HTTPRequest buildAccessTokenRequestWithJwtAuthenticationAndAuthorization
-                            clientAuthClaimsSet,
-                            signerFactory.getSigner(),
-                            configService.enabled(KID_JAR_HEADER));
+                    JwtHelper.createSignedJwt(clientAuthClaimsSet, signerFactory.getSigner(), true);
```
I think the `true` can to be removed here. If its always `true` we can remove the `boolean includeKid` from the method?

undermad replied:
StoredIdentityService.java actually use this method - set to false - and doesn't depend on the feature flag.
https://github.com/govuk-one-login/ipv-core-back/blob/main/libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/evcs/service/StoredIdentityService.java#L97

Joe-Edwards-GDS replied:
That sounds like a mistake to me - IMO we should always be including a `kid` (otherwise it's impossible to keep track of the key used)

undermad replied:
I removed this boolean value then and change StroredIdentityService to include kid in jwt header.

AmritSidhu reviewed (commented):
`kidJarHeaderEnabled` can be removed from the `/local-running/core.local.params.yaml` file too

undermad committed:
PYIC-8327: Remove includeKid boolean from createSignedJwt and include kid value in StoreIdentityService.

Method parameter passed to createSignedJwt was set incorrectly to false in StoredIdentityService.java, as kid should be included.
